lib/features/support/logger.rb

```diff
@@ -18,3 +18,44 @@ def initialize
 end
 
 $logger = MazeLogger.instance
+
+# A collection of logging utilities
+class LogUtil
+  class << self
+    # Logs requests and other data, accounting for things file upload requests that are too big to log meaningfully.
+    #
+    # @param severity [Integer] A Logger::Severity
+    # @param data [Hash] The data to log (currently needs to be a Hash)
+    def log(severity, data)
```

Comment:
Potentially confusing name - I would have expected this to handle a string for normal logging.

```suggestion
    def log_hash(severity, data)
```

Reply:
Interesting point - I initially called it `log_data` in the expectation that it will evolve to handle more than just hashes.  I changed it to `log`, I think because of my Java experience and method overloading - which I don't think applies in Ruby.  So on reflection I'll change it to `log_hash` as you suggest and change it again later if necessary.  It may only ever log hashes.